vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp

```diff
@@ -892,7 +895,15 @@ void WindowsTextInputComponentView::OnCharacterReceived(
   }
   emitter->onKeyPress(onKeyPressArgs);
 
-  WPARAM wParam = static_cast<WPARAM>(args.KeyCode());
+  WPARAM wParam;
+
+  // Auto-capitalize when character is alphabetic AND current string index qualifies for capitalization.
+  if (std::isalpha(args.KeyCode()) && shouldAutoCapitalize()) {
```

Comment:
OnCharacterReceived is only one of many different ways that the text within the TextInput can be modified.
I dont think IME input, copy/paste, modification using UIA, etc will hit this code.

Instead of trying to modify the input to get the text you want, maybe we can look for [EN_UPDATE](https://learn.microsoft.com/en-us/windows/desktop/Controls/en-update) which will get triggered for any method of the text getting changed, and then use [ITextRange::ChangeCase (tom.h) - Win32 apps | Microsoft Learn](https://learn.microsoft.com/en-us/windows/win32/api/tom/nf-tom-itextrange-changecase) to modify the case of the text in the TextInput?

Reply:
What is UIA?

Reply:
UI Automation
https://learn.microsoft.com/en-us/windows/win32/winauto/entry-uiauto-win32

Reply:
I tried calling ChangeCase on EN_UPDATE... I noticed that whenever auto-capitalization occurs, there's a side effect where the caret is moved back to the beginning of the text. Should I just manually move the caret back to the right index?

In the other hand, tomSentenceCase seems to capitalize only after `.`, but not after `?` or `!`.

Reply:
I'm scoping the changes to only words for now, created issue to track implementing sentences/words: #13479